tests/rbd/test_scale_group_snap_clone.py

```diff
@@ -0,0 +1,241 @@
+    """
+    kw["client"] = client
+    rbd = rbd_obj.get("rbd")
+    rbd1 = Rbd(kw["client"])
+    fio = kw.get("config", {}).get("fio", {})
+
+    for pool_type in rbd_obj.get("pool_types"):
+        rbd_config = kw.get("config", {}).get(pool_type, {})
+        multi_pool_config = getdict(rbd_config)
+        for pool, pool_config in multi_pool_config.items():
+            log.info(f"All pool details : {pool} and {pool_config}")
+            if "data_pool" in pool_config.keys():
+                _ = pool_config.pop("data_pool")
+
+            # Create a group per pool
+            log.info(f"Create a group {pool}")
+            group = kw.get("config", {}).get("group", "image_group_default")
+            group_create_kw = {"client": client, "pool": pool, "group": group}
+            rc = create_group_and_verify(**group_create_kw)
+            if rc:
+                log.error(f"group {group} creation failed, test case fail")
+                return 1
+            else:
+                log.info("STAGE: group creation succeeded")
+
+            for image, image_config in pool_config.items():
+                log.info(f"All image details : {image} and {image_config}")
+
+                # Add all 100+ images to the group using a loop.
+                log.info(f"Adding image: {image} in group {group}")
+                add_image_group_kw = {
+                    "client": client,
+                    "pool": pool,
+                    "group": group,
+                    "image": image,
+                }
+                rc = add_image_to_group_and_verify(**add_image_group_kw)
+                if rc:
+                    log.error(f"Image {image} add failed, test case fail")
+                    return 1
+                else:
+                    log.info(f"STAGE: image {image} add succeeded")
+
+            # Take a snapshot of the group using the rbd group snap create command.
+            group_snap = kw.get("config", {}).get("group_snap", "group_snap_default")
+            log.info(f"Creating group snapshot: {group_snap}")
+            snap_create_kw = {
+                "client": client,
+                "pool": pool,
+                "group": group,
+                "snap": group_snap,
+            }
+            rc = create_snap_and_verify(**snap_create_kw)
+            if rc:
+                log.error(f"snap {group_snap} create failure")
+                return 1
+            else:
+                log.info(f"STAGE: snap {group_snap} creation with validation done")
+
+            # Verify the group snapshot exists using the rbd group snap ls command.
+            log.info(f"Verify group id for group: {group} snap: {group_snap}")
+            group_ls_kw = {
+                "client": client,
+                "pool": pool,
+                "group": group,
+                "format": "json",
+            }
+            (g_ls_out, _) = group_info(**group_ls_kw)
```

Comment:
please keep the variable more descriptive here as well `group_output`

Reply:
Done